### PR TITLE
Headerbar buttons & API Bar refactor

### PR DESCRIPTION
**Pull request by robertkowalski**
The PR is divided into 4 commits with different sub themes, each of them has a commit message that describe the current change:

1. https://github.com/robertkowalski/couchdb-fauxton/commit/62d872691d3f9993101b274ca1162b011c717b2a

2. https://github.com/robertkowalski/couchdb-fauxton/commit/6dce8cae0495a25a1b3b4bd7bd90e390bc4adef0

3. https://github.com/robertkowalski/couchdb-fauxton/commit/fb2d74a8888d9862532e1e36e31a39b3945bf598

4. https://github.com/robertkowalski/couchdb-fauxton/commit/66442eef4e4184856db66858b721a604937e4cbb

The code is covered by many of our selenium tests, some of them needed small adjustments

### CODE REVIEW

File: app/addons/components/react-components.react.jsx [hunk 1]
```diff
@@ -14,20 +14,33 @@ define([
...
   'ace/ace',
   'plugins/beautify'
 ],
 
-function (app, FauxtonAPI, React, Components, ace, beautifyHelper) {
+function (app, FauxtonAPI, React, FauxtonComponents, ace, beautifyHelper) {
   var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
 
   var ToggleHeaderButton = React.createClass({
+    getDefaultProps: function () {
```
Comment (robertkowalski) [inline app/addons/components/react-components.react.jsx]: @benkeen added!

**Comment by robertkowalski** [comment]
@benkeen adressed the comments, had to rebase as i had a merge conflict with your recent header-flexbox work and github lost all comments. :(

here is the other comment i answered: https://github.com/robertkowalski/couchdb-fauxton/commit/6dce8cae0495a25a1b3b4bd7bd90e390bc4adef0#commitcomment-13490635

**Comment by robertkowalski** [comment]
@benkeen +1 ?

**Comment by benkeen** [comment]
Gimme just a few more mins + I'll give it one more look over :)

**Comment by benkeen** [comment]
Couple of minor style things:
- there used to be more space between the icons and text for "Add New Database" and "API URL" 
- the icons are a little too high now; they used to be horizontally aligned with the text. 
screen shot 2015-09-29 at 10 36 42 am

- Same for Query Options on the Database pages. 
screen shot 2015-09-29 at 10 38 21 am

**Comment by benkeen** [comment]
Other than those, looks good! +1 with the style fixes.

**Comment by robertkowalski** [comment]
@benkeen fixed the icon position, good eye!

merged as:

ebd9ced89d27de297526acf74ebad88f1c6a6711
685d6f82d60f7d020c64bdf5010dc1f84d77bf5e
d1d98323d5145af0129ba84c28a5b567f409044d
69b686e31cc0ea9f4c7eeb254dc061447995f2c9

**Pull request closed without merging**